xcdat/temporal.py

```diff
@@ -1343,7 +1355,7 @@ def _convert_df_to_dt(self, df: pd.DataFrame) -> np.ndarray:
                 df_new[component] = default_val
 
         dates = [
-            cftime.datetime(year, month, day, hour)
+            self.date_type(year, month, day, hour)
```

Comment:
The `date_type` is used to enable xarray's indexing/subsetting with `cftime` objects.